src/pages/ReportWelcomeMessagePage.js

```diff
@@ -0,0 +1,94 @@
+import React, {useCallback, useRef, useState} from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {View} from 'react-native';
+import ExpensiMark from 'expensify-common/lib/ExpensiMark';
+import compose from '../libs/compose';
+import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
+import ScreenWrapper from '../components/ScreenWrapper';
+import Navigation from '../libs/Navigation/Navigation';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import styles from '../styles/styles';
+import reportPropTypes from './reportPropTypes';
+import withReportOrNotFound from './home/report/withReportOrNotFound';
+import Text from '../components/Text';
+import TextInput from '../components/TextInput';
+import * as Report from '../libs/actions/Report';
+import ONYXKEYS from '../ONYXKEYS';
+import CONST from '../CONST';
+import FullPageNotFoundView from '../components/BlockingViews/FullPageNotFoundView';
+import Form from '../components/Form';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+
+    /** The report currently being looked at */
+    report: reportPropTypes.isRequired,
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** Report ID passed via route r/:reportID/welcomeMessage */
+            reportID: PropTypes.string,
+        }),
+    }).isRequired,
+};
+
+function ReportWelcomeMessagePage(props) {
+    const parser = new ExpensiMark();
+    const [welcomeMessage, setWelcomeMessage] = useState(parser.htmlToMarkdown(props.report.welcomeMessage));
+    const welcomeMessageInputRef = useRef(null);
+
+    const handleWelcomeMessageChange = useCallback((value) => {
+        setWelcomeMessage(value);
+    }, []);
+
+    const submitForm = useCallback(() => {
+        Report.updateWelcomeMessage(props.report.reportID, props.report.welcomeMessage, welcomeMessage);
+    }, [props.report.reportID, props.report.welcomeMessage, welcomeMessage]);
+
+    return (
+        <ScreenWrapper
+            onTransitionEnd={() => {
+                if (!welcomeMessageInputRef.current) {
+                    return;
+                }
+                welcomeMessageInputRef.current.focus();
```

Comment:
Hi ✋ Coming from https://github.com/Expensify/App/issues/21523

Since the input is multiline and the value can be set from the previous message saved, it makes sense to focus the input and put the cursor on to the end of the text.